src/components/container/ConnectDevicesMenu/ConnectDevicesMenu.tsx

```diff
@@ -23,34 +24,50 @@ export interface ConnectDevicesMenuProps {
     previewState?: "iOS" | "Android" | "Web" | "ConnectedStates"
     headerVariant?: "large" | "medium"
     connectExternalAccountOptions?: ConnectExternalAccountOptions
+    enableAppleHealthSurvey?: string
+    enableGoogleFitSurvey?: string
 }
 
 export default function (props: ConnectDevicesMenuProps) {
     const [loading, setLoading] = useState(true);
     const [settings, setSettings] = useState<DataCollectionSettings | null>(null);
     const [deviceExternalAccounts, setDeviceExternalAccounts] = useState<ExternalAccount[] | null>(null);
     const [platform, setPlatform] = useState<string | null>(null);
+    const [hasRecentAppleHealthData, setHasRecentAppleHealthData] = useState<boolean>(false);
 
     function initialize() {
         if (props.previewState) {
-            setSettings(previewSettings);
+            let settings = { ...previewSettings };
+            if (props.enableAppleHealthSurvey || props.enableGoogleFitSurvey) {
+                settings.queryableDeviceDataTypes = [];
+            }
+            setSettings(settings);
 
             setDeviceExternalAccounts([]);
+
+            setPlatform("Web");
             if (props.previewState == "ConnectedStates") {
                 setDeviceExternalAccounts(previewAccounts);
+                setPlatform("iOS");
+                setHasRecentAppleHealthData(true);
             }
 
-            setPlatform("Web");
             if (props.previewState == "iOS") {
```

Comment:
This can be an `else if` off the condition above.

Reply:
Done